src/main/java/com/customer/controller/CustomerController.java

```diff
@@ -24,31 +29,49 @@
 	private CustomerServiceImpl customerServiceImpl;
 
 	@Autowired
-	CustomerController(CustomerServiceImpl customerServiceImpl)
-	{
+	CustomerController(CustomerServiceImpl customerServiceImpl) {
 		this.customerServiceImpl = customerServiceImpl;
 	}
 
 	// get the list of all the customers
 	@GetMapping("/customers")
 	public List<CustomerResponse> getAllCustomers() {
-
 		return customerServiceImpl.getAllCustomers();
-
 	}
 
 	/* Post request */
-	@PostMapping("/customers")
+	@PostMapping("/customerRegister")
 	public String createCustomer(@Valid @RequestBody Customer customer) {
-		customerServiceImpl.createCustomer(customer);
+		try {
+			customerServiceImpl.createCustomer(customer);
+		} catch (Exception e) {
+			System.out.print("Values can't null");
+		}
 		return "Customer is created successfully";
 
 	}
 
 	/* get the customer by Id */
 	@GetMapping("/customers/{id}")
-	public @ResponseBody CustomerResponse findCustomerById(@PathVariable int id) {
-		return customerServiceImpl.findCustomerById(id);
+	public @ResponseBody CustomerResponse findCustomerById(@PathVariable int id, Principal principal) {
+		return customerServiceImpl.findCustomerById(id, principal);
+	}
+
+	/* update the customer details */
+	@PutMapping("/customers/{id}")
+	public ResponseEntity<String> updateCustomer(@PathVariable int id, @RequestBody Customer customer,
+			Principal principal) {
+
+		customerServiceImpl.updateCustomerDetails(id, customer, principal);
+
+		return new ResponseEntity<String>("User is updated successfully", HttpStatus.OK);
```

Comment:
this could be `ResponseEntity.ok(` like you are using in the JWT Controller class

Reply:
Done